### DISCUSSION TITLE
Weird situation...

### DISCUSSION THREAD

**Discussion by cduytran** [pocketbase/pocketbase#1829] [Q&A]
I have two collections:

-packs with a field `isActive = boolean`
-eod with a relation field. `eodPackRef`

I'd like to create a query that return all packs that where packs.isActive = true AND packs.id is not in collections eod that is created today.

So far I have to make 2 queries. One is to get all packs.isActive.
Then another query to get all eod that isn't created today.
Finally, I filter out any pack that doesn't meet my requirements.

```
//get all active packs
const activePacks = await pb.collection('packs').getFullList( 200, { filter: `isActive = true`, } )
```

```
//get all eod that is created today
const todayEods = await pb.collection('eod').getFullList( 200, {filter: `created = "${today}"` ,  } )
```

```
activePacks.forEach ( pack ){
 todayEods.forEach ( eod ) {
  if ( pack.id !== eod.eodPackRef[0] ){
    //if pack doesn't have eod that is created today, we save it
    savedPacks.push(pack)
  }
 }
}
```

Thanks for any suggestions.

**Comment by ganigeorgiev** [comment]
I don't think there is a more simpler way at the moment.
The above approach seems OK to me. I'm not sure only why the 0 index of `eodPackRef` is checked, but if it is multi-`relation` field, aka. Array, then you can do something like `!eod.eodPackRef.includes(pack.id)` if you want to check for existence. Here is a slightly shorter version that also replaces forEach with `filter` and `find`:

```js
const savedPacks = activePacks.filter((pack) => {
  return !!todayEods.find((eod) => !eod.eodPackRef.includes(pack.id));
})
```

_Indirect relationa filter and sort is in the [roadmap](https://github.com/orgs/pocketbase/projects/2) and will eventually simplify the above in 1 request, but it comes with some performance overhead and I'll have to test it more thoroughly before enabling it._

**Reply by cduytran** [reply]
Thanks.

I have an unrelated question regarding multi select relation. I am not quite sure what the Max Select does.

If Max Select is 1, does that mean the relation field can only hold one value? And if Max Select is 2, then it means that relation field is an array of 2 ?  And if Max Select is an array, how do we manage to create/update/delete a particular relation field (pb.collection('somecollections').update('someId', data) where data = { relationField [ someIndex ] = 'something' } ?

Currently, my Max Select is empty (or zero). does that mean I can have unlimited array? <-- def, not intended by me.

**Reply by cduytran** [reply]
!!todayEods.find((eod) => !eod.eodPackRef.includes(pack.id));

Don't quite understand the !! syntax..;-)

**Reply by ganigeorgiev** [reply]
If "Max select = 1" then this means that it can hold only 1 value (eg. `"0gt8327uh79a8y5"`).
If "Max select != 1" then this means that it can hold multiple values (eg. ["0gt8327uh79a8y5", "utgp2qzuzudz27f"]).

> Currently, my Max Select is empty (or zero). does that mean I can have unlimited array?

Yes. _Please note that currently when changing the "Max select" constraint we are not updating the already created records (this may change in the future; see also https://github.com/pocketbase/pocketbase/discussions/1808#discussioncomment-4898464)._

> how do we manage to create/update/delete a particular relation field

You can append or remove a single value from a multi-`relation` field by using the `+` and `-` field modifiers:

```js
await pb.collection("example").update("RECORD_ID", {
    "someRelField-": ["id2", "id3"], // removes id2 and id3
    "someRelField+": ["id5", "id6"], // add id5 and id6
    
    // original value ["id1", "id2", "id3", "id4"]
    // after the call ["id1", "id4", "id5", "id6"]
})

// or alternatively you can also just pass the entire final modified array (this allow individual sorting)
await pb.collection("example").update("RECORD_ID", {
    "someRelField": ["id1", "id4", "id5", "id6"]
})
```

**Reply by ganigeorgiev** [reply]
> Don't quite understand the !! syntax..;-)

This is just double negation used as a short "normalization" syntax to ensure that the returned value will be always boolean (it is not required in the above):

```js
!!undefined     // false
!!{"test": 123} // true
```
